dewy/documents/router.py

```diff
@@ -25,15 +25,25 @@ async def add_document(
 ) -> Document:
     """Add a document."""
 
-    row = None
+    collection_id = req.collection_id    
     async with pg_pool.acquire() as conn:
+        if collection_id is None:
+            collection_id = await conn.fetchval(
```

Comment:
Would this need to be in a transaction? Alternatively, we could write a single query that retrieves the ID and sets it, and then use that (which may be simpler than pushing the logic into Python).

Also, shouldn't this be looking at the collection name (so selecting from `collection` rather than `document`)?